internal/tracing/tracing.go

```diff
@@ -20,3 +22,16 @@ func RecordError(span trace.Span, err error, description string) {
 	span.RecordError(err)
 	span.SetStatus(codes.Error, description)
 }
+
+func SpanContextFromHeaders(res *http.Response) trace.SpanContext {
+	traceIDstr := res.Header.Get(HeaderFlyTraceId)
+	spanIDstr := res.Header.Get(HeaderFlySpanId)
+
+	traceID, _ := trace.TraceIDFromHex(traceIDstr)
+	spanID, _ := trace.SpanIDFromHex(spanIDstr)
```

Comment:
Just checking: if the trace ID or span ID is invalid, what will the span be linked to?

Reply:
nothing. both need to be valid, to have a link. it won't cause any problems :)

Reply:
👍. BTW I probably should have been more specific: I assume that if there's an error then `traceID`/`spanID` will be the zero values for the types, so I just wanted to make sure that that's okay, versus explicitly handling the errors.